src/Tribe/Modules/Single_Event_Template.php

```diff
@@ -0,0 +1,169 @@
+
+		if ( ! in_array( $data->meta['origin'], $valid_origins ) ) {
+			return $event;
+		}
+
+		$valid_sources = [];
+
+		/**
+		 * Allows filtering the source URLs to which the template should be applied to.
+		 * If you want to limit this to a certain source URL, set it here.
+		 *
+		 * @var array $valid_sources The array of the import URLs
+		 */
+		$valid_sources = apply_filters( 'tribe_ext_ea_additional_options_valid_sources', $valid_sources );
+
+		if (
+			! empty( $valid_sources )
+			&& ! in_array( $data->meta['source'], $valid_sources )
+		) {
+			return $event;
+		}
+
+		// Get the template from the saved setting.
+		$template_post_id = tribe_get_option( Settings::PREFIX . 'default_template', false );
+
+		if ( $template_post_id == 'enable' ) {
+			$blocks = [
+				'datetime'       => '<!-- wp:tribe/event-datetime /-->',
+				'featured_image' => '<!-- wp:tribe/featured-image /-->',
+				'content_start'  => '<!-- wp:paragraph {"placeholder":"Add Description..."} -->',
+				'content'        => '<p>[tec_ea_content]</p>',
+				'content_end'    => '<!-- /wp:paragraph -->',
+				'organizer'      => '<!-- wp:tribe/event-organizer /-->',
+				'venue'          => '<!-- wp:tribe/event-venue /-->',
+				'sharing'        => '<!-- wp:tribe/event-links /-->',
+				'related'        => '<!-- wp:tribe/related-events /-->',
+				'comments'       => '<!-- wp:post-comments-form /-->',
+			];
+
+			$template = implode( "\n", $blocks );
+		} else {
+			$post = get_post( $template_post_id );
+
+			// If there is no template, or not the right format then skip.
+			if (
+				! $post
+				|| $post->post_type != 'tribe_events'
+				|| $post->post_status != 'draft'
+			) {
+				return $event;
+			}
+
+			// Get the post content template.
+			$template = $post->post_content;
+		}
+
+		// The post content from the source site.
+		$description_from_source = $event['post_content'];
+
+		// If there is an organizer in the source, then change it in the template.
+		if ( isset( $event['Organizer'] ) ) {
+			$template = str_replace(
+				'<!-- wp:tribe/event-organizer /-->',
+				'<!-- wp:tribe/event-organizer {"organizer":' . $event['Organizer']['OrganizerID'][0] . '} /-->',
+				$template
+			);
+		}
+
+		// Convert line breaks to paragraphs.
+		$search_line_breaks      = [
+			'(\n\n)',
+			'(\n)',
+		];
+		$replace_line_breaks     = [
+			'</p><!-- /wp:paragraph --><!-- wp:paragraph --><p>',
+			'<br>',
+		];
+		$description_from_source = preg_replace(
+			$search_line_breaks,
+			$replace_line_breaks,
+			$description_from_source
+		);
+
+		// Replace the placeholder with the content from the source.
+		$template = str_replace(
+			'[tec_ea_content]',
+			$description_from_source,
+			$template,
+		);
+
+		// Replace double paragraphs, just in case.
+		$s        = [
+			'<p><p>',
+			'</p></p>',
+		];
+		$r        = [
+			'<p>',
+			'</p>',
+		];
+		$template = str_replace( $s, $r, $template );
```

Comment:
You can just use a single `str_replace` call using array for search and replace; order matters.

Reply:
✔️ Combined the string replacements into one.